checker/src/org/checkerframework/checker/nullness/NullnessAnnotatedTypeFactory.java

```diff
@@ -507,8 +507,26 @@ public NullnessTypeAnnotator(InitializationAnnotatedTypeFactory<?, ?, ?, ?> atyp
 
     @Override
     public AnnotationMirror getFieldInvariantAnnotation() {
-        Elements elements = processingEnv.getElementUtils();
-        return AnnotationUtils.fromClass(elements, NonNull.class);
+        return NONNULL;
+    }
+
+    /**
+     * Returns whether or not field has the invariant annotation.
+     *
+     * <p>If the field is a type variable, this method return false if any possible instantiation of
+     * the type parameter could have the invariant annotation. In other words, is the lower
+     * bound @NonNull?
+     *
+     * @param field field that might have invariant annotation
+     * @return whether or not field has the invariant annotation
+     */
+    @Override
+    protected boolean hasInvariantAnnotation(VariableTree field) {
```

Comment:
This is only the implementation for systems where the invariant annotation is bottom.  If it is some other annotation, then a type is has the invariant annotation if the lower bound annotation is a subtype of (or equal to) the invariant annotation and the upper bound is a super type of (or equal to) the annotation.  

By making this method abstract, it forces future subclasses to implement this method.